hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestDirectoryDeletingServiceV1.java

```diff
@@ -0,0 +1,273 @@
+    volumeName = bucket.getVolumeName();
+    bucketName = bucket.getName();
+
+    String rootPath = String.format("%s://%s.%s/",
+        OzoneConsts.OZONE_URI_SCHEME, bucketName, volumeName);
+
+    // Set the fs.defaultFS and start the filesystem
+    conf.set(CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY, rootPath);
+    // Set the number of keys to be processed during batch operate.
+    conf.setInt(OZONE_FS_ITERATE_BATCH_SIZE, 5);
+
+    fs = FileSystem.get(conf);
+  }
+
+  @AfterClass
+  public static void teardown() {
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+    IOUtils.closeQuietly(fs);
+  }
+
+  @After
+  public void cleanup() {
+    try {
+      Path root = new Path("/");
+      FileStatus[] fileStatuses = fs.listStatus(root);
+      for (FileStatus fileStatus : fileStatuses) {
+        fs.delete(fileStatus.getPath(), true);
+      }
+    } catch (IOException ex) {
+      fail("Failed to cleanup files.");
+    }
+  }
+
+  @Test
+  public void testDeleteEmptyDirectory() throws Exception {
+    Path root = new Path("/rootDir");
+    Path appRoot = new Path(root, "appRoot");
+    fs.mkdirs(appRoot);
+
+    Table<String, OmKeyInfo> deletedDirTable =
+        cluster.getOzoneManager().getMetadataManager().getDeletedDirTable();
+    Table<String, OmDirectoryInfo> dirTable =
+        cluster.getOzoneManager().getMetadataManager().getDirectoryTable();
+
+    // Before delete
+    GenericTestUtils.waitFor(() ->
+            assertTableRowCount(0, deletedDirTable),
+        1000, 120000); // 2 minutes
+
+    GenericTestUtils.waitFor(() ->
+            assertTableRowCount(2, dirTable),
+        1000, 120000); // 2 minutes
+
+    // Delete the appRoot, empty dir
+    fs.delete(appRoot, true);
+
+
+    // After Delete
+    checkPath(appRoot);
+
+    GenericTestUtils.waitFor(() ->
+            assertTableRowCount(0, deletedDirTable),
+        1000, 120000); // 2 minutes
+
+    GenericTestUtils.waitFor(() ->
+            assertTableRowCount(1, dirTable),
+        1000, 120000); // 2 minutes
+
+    Assert.assertTrue(dirTable.iterator().hasNext());
+    Assert.assertEquals(root.getName(),
+        dirTable.iterator().next().getValue().getName());
+
+    DirectoryDeletingService dirDeletingService =
+        (DirectoryDeletingService) cluster.getOzoneManager().getKeyManager()
+            .getDirDeletingService();
+
+    GenericTestUtils.waitFor(
+        () -> dirDeletingService.getDeletedDirsCount() >= 1,
+        1000, 10000);
+    GenericTestUtils.waitFor(
+        () -> dirDeletingService.getMovedFilesCount() >= 0,
+        1000, 10000);
+
+    Assert.assertTrue(dirDeletingService.getRunCount() > 1);
+  }
+
+  @Test
+  public void testDeleteWithMultiLevels() throws Exception {
+    Path root = new Path("/rootDir");
+    Path appRoot = new Path(root, "appRoot");
+
+    for (int i = 1; i <= 3; i++) {
+      Path parent = new Path(appRoot, "parentDir" + i);
+      Path child = new Path(parent, "childFile");
```

Comment:
Can we reset the ozone.path.deleting.limit.per.task to a small value and create dir/child files more than that value? So that we could test current purge logic works fine with the delete limitation per task.

Reply:
Added new tests covering multiple batch deletion including dirs/files.